src/Data/Text/Internal/Encoding.hs

```diff
@@ -0,0 +1,522 @@
+{-# LANGUAGE BangPatterns, CPP, GeneralizedNewtypeDeriving, MagicHash,
+    UnliftedFFITypes #-}
+{-# LANGUAGE Trustworthy #-}
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE ViewPatterns #-}
+
+-- |
+-- Internals of "Data.Text.Encoding".
+--
+-- If you'd like to depend on something from here that's not in "Data.Text.Encoding",
+-- please request to have it exported!
+module Data.Text.Internal.Encoding where
```

Comment:
Please add `@since 2.0.2` to haddock header. Please use an explicit export list. 

Also let's slap a warning about instability / non-adherence to PVP as in other internal modules.